packages/langium/src/serializer/json-serializer.ts

```diff
@@ -53,7 +54,10 @@ export class DefaultJsonSerializer implements JsonSerializer {
     }
 
     serialize(node: AstNode, options?: JsonSerializeOptions): string {
-        return JSON.stringify(node, (key, value) => this.replacer(key, value, options?.refText), options?.space);
+        const defaultReplacer = (key: string, value: unknown) => this.replacer(key, value, options?.refText);
```

Comment:
While we're at it, could you change the third parameter of the `replacer` method to the JsonSerializeOptions object so we can extend it more easily in the future?

Reply:
I had that in mind, too, need to sync with #886 (generator) as there will be a conflict.